src/math/Vector3.js

```diff
@@ -5,87 +5,264 @@
  * @author mikael emtinger / http://gomo.se/
  * @author egraether / http://egraether.com/
  * @author WestLangley / http://github.com/WestLangley
+ * @author Ben Houston / https://clara.io
  */
 
-THREE.Vector3 = function ( x, y, z ) {
+ THREE.Vector3 = function ( array, x, y, z ) {
 
-	this.x = x || 0;
-	this.y = y || 0;
-	this.z = z || 0;
+	 if( !( ( array instanceof Float32Array ) || ( array === undefined ) ) ) {
```

Comment:
`typeof array === 'number'` would actually be better, I think. Then can also go 64-bit (or even integers or JS `Array`) when you want.